conseil-common/src/main/scala/tech/cryptonomic/conseil/common/rpc/RpcClient.scala

```diff
@@ -0,0 +1,75 @@
+package tech.cryptonomic.conseil.common.rpc
+
+import scala.util.control.NoStackTrace
+
+import cats.effect.Concurrent
+import com.typesafe.scalalogging.LazyLogging
+import fs2.Stream
+import org.http4s.client._
+import org.http4s.client.dsl.Http4sClientDsl
+import org.http4s.{EntityDecoder, EntityEncoder, Method, Uri, Header}
+
+import tech.cryptonomic.conseil.common.rpc.RpcClient._
+
+class RpcClient[F[_]: Concurrent](
+    endpoint: String,
+    maxConcurrent: Int,
+    client: Client[F],
+    headers: Header*
+) extends Http4sClientDsl[F]
+    with LazyLogging {
+
+  def stream[P, R](batchSize: Int)(request: Stream[F, RpcRequest[P]])(
+      implicit decode: EntityDecoder[F, Seq[RpcResponse[R]]],
+      encode: EntityEncoder[F, List[RpcRequest[P]]]
+  ): Stream[F, R] =
+    request
+      .chunkN(batchSize)
+      .evalTap { requests =>
+        Concurrent[F].delay(logger.info(s"Call RPC in a batch of: ${requests.size}"))
+
+      }
+      .mapAsyncUnordered(maxConcurrent) { chunks =>
+        client
+          .expect[Seq[RpcResponse[R]]](
+            Method.POST(chunks.toList, Uri.unsafeFromString(endpoint), headers: _*)
+          )
+      }
+      .flatMap(Stream.emits)
+      .flatMap {
+        case RpcResponse(_, _, Some(error), _) =>
+          Stream.raiseError[F](error)
+        case RpcResponse(_, Some(result), _, _) =>
+          Stream(result)
+        case _ =>
+          Stream.raiseError[F](
+            new UnsupportedOperationException(
+              "Empty response result as well as error information."
```

Comment:
The log message is not totally clear to me.

Reply:
it's a hypothetical situation when RPC server returns an empty response and no error message, this is probably not possible

Reply:
Yes, I see what you mean, I was trying to convey that the message itself seems not too self-explanatory, like, if I read that on a log entry, I won't necessarily understand what's wrong.